frontend/src/chrome/IconLink.tsx

```diff
@@ -1,21 +1,42 @@
 import React from 'react'
 import ExternalLink from './ExternalLink'
-import Icon from './Icon'
+import Icon, { IconSize } from './Icon'
 
 interface IconLinkProps {
   icon: string
-  link: string
+  link?: string
+  onClick?: () => void
+  size?: IconSize
 }
 
 const IconLink: React.FC<IconLinkProps> = ({
   icon,
-  link
-}) => (
-  <div className='ml-2'>
-    <ExternalLink to={link} className='text-qrinavy hover:text-qrinavy-600'>
-      <Icon icon={icon} size='sm'/>
-    </ExternalLink>
-  </div>
-)
+  link,
+  onClick,
+  size='sm'
+}) => {
+  const iconComponent = <Icon icon={icon} size={size} />
```

Comment:
I don't think this const is necessary, better to just copy the component in both branches